src/main/java/org/opendevstack/provision/services/JiraAdapter.java

```diff
@@ -108,9 +123,28 @@ public ProjectData createJiraProjectForProject(ProjectData project,
           project.admins.add(new BasicUser(null, project.admin, project.admin));
       }
       
-      FullJiraProject created = this
-          .callJiraCreateProjectApi(this.buildJiraProjectPojoFromApiProject(project),
-              crowdCookieValue);
+      FullJiraProject toBeCreated = 
+    		  this.buildJiraProjectPojoFromApiProject(project);
+      
+      FullJiraProject created = null;
+      try {
+    	  created = this.callJiraCreateProjectApi(toBeCreated,crowdCookieValue); 
+      } catch (IOException jiracreateException) 
+      {
+          logger.debug("error creating project with template {}: {}", 
+        		  toBeCreated.projectTemplateKey,
+        		  jiracreateException.getMessage());
+    	  if (jiracreateException.getMessage() != null && jiracreateException.getMessage().startsWith("400")) {
+    		  toBeCreated.projectTypeKey = jiraTemplateType;
+    		  toBeCreated.projectTemplateKey = jiraTemplateKey;
+        	  created = this.callJiraCreateProjectApi(toBeCreated,crowdCookieValue);
+        	  project.projectType = defaultProjectKey;
```

Comment:
If I understand this correctly, it creates the project with the default template if the original request fails with a 400. Can you explain *why* we are doing this?

At least we should log on INFO level I'd say.

Reply:
@michaelsauter  - while on confluence you can check for the blueprint - on jira there is NO rest api to make this work, so we have to go kind of trial and error - and HTTP 400 is returned only in this case - so we fall back then to the default one, which we know works. makes sense?

Reply:
Yes makes sense. Maybe we can add this as a comment to the code and log on info level to make this more obvious to readers/users.

Reply:
@michaelsauter  - -info added now. check the new push.